### PR TITLE
feat(save-and-load-assessment): add save assessment factory

### PR DESCRIPTION
**Pull request by shanisebarona**
#### Description of changes
This PR adds the save assessment factory and removes the save assessment button from rendering in FastPass. 

#### Pull request checklist

- [x] Addresses an existing issue: Feature request #653
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

FastPass:
image

Assessments:
image

### CODE REVIEW

File: src/DetailsView/components/save-assessment-factory.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export type SaveAssessmentFactoryDeps = {
+    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
+};
+
+export type SaveAssessmentFactoryProps = CommandBarProps & {
+    deps: SaveAssessmentFactoryDeps;
+};
+
+export function getReportForAssessment(props: SaveAssessmentFactoryProps): JSX.Element {
```
Comment (jalkire) [inline src/DetailsView/components/save-assessment-factory.tsx:16]: You might want to consider a different name for this and the below function--maybe something more related to what it's returning, like "getSaveButtonFor<Assessment or Fastpass>"?

File: src/DetailsView/components/save-assessment-factory.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { CommandBarProps } from 'DetailsView/components/details-view-command-bar';
+import { DetailsViewActionMessageCreator } from '../actions/details-view-action-message-creator';
+import { SaveAssessmentButton } from 'DetailsView/components/save-assessment-button';
+
+export type SaveAssessmentFactoryDeps = {
+    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
+};
+
+export type SaveAssessmentFactoryProps = CommandBarProps & {
+    deps: SaveAssessmentFactoryDeps;
```
Comment (jalkire) [inline src/DetailsView/components/save-assessment-factory.tsx:13]: It looks like you aren't using these deps or props, so you'd be fine to take them out if you wanted. I know you'll probably need them later, though

File: src/DetailsView/components/save-assessment-factory.tsx (hunk 1 above)
Comment (shanisebarona) [reply src/DetailsView/components/save-assessment-factory.tsx]: Updated, thank you!

File: src/DetailsView/components/save-assessment-factory.tsx (hunk 2 above)
Comment (shanisebarona) [reply src/DetailsView/components/save-assessment-factory.tsx:10]: I think I'll leave it for now, since I'll need it for the next PR.

**Review by jalkire — APPROVED** [review]
👍

**Pull request merged**